### PR TITLE
Fix svn provider

### PR DESCRIPTION
**Pull request by odcinek**
'MM' and ' M' statuses were breaking badly
gd integration tests from master passed ok

### CODE REVIEW

File: lib/between_meals/repo/svn.rb [hunk 1]
```diff
@@ -91,6 +96,23 @@ def valid_ref?(ref)
...
         raise Changeset::ReferenceError
       end
+
+      private
+
+      def parse_status(changes)
+        # http://svnbook.red-bean.com/en/1.0/re26.html
+        changes.lines.map do |line|
+          case line
+          when (/^\s?(\w)\w?\s+(\S+)$/)
```
Comment (elthariel) [inline lib/between_meals/repo/svn.rb]: /^([\w ])\w?\s+(\S+)$/

File: lib/between_meals/repo/svn.rb [hunk 2]
```diff
@@ -91,6 +96,23 @@ def valid_ref?(ref)
...
         raise Changeset::ReferenceError
       end
+
+      private
+
+      def parse_status(changes)
+        # http://svnbook.red-bean.com/en/1.0/re26.html
+        changes.lines.map do |line|
+          case line
+          when (/^([\w ])\w?\s+(\S+)$/)
```
Comment (jaymzh) [inline lib/between_meals/repo/svn.rb]: If the first letter is a space, you want the second letter which you won't capture here... I thnk?

File: spec/svn_spec.rb [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    {
+      :name => 'handle attribute modifications',
+      :changes => ' M foo/bar/baz',
+      :result => [
+        { :status => :modified, :path => 'bar/baz' },
+      ],
+    },
+    {
+      :name => 'handle file & attribute modifications',
+      :changes => 'MM foo/bar/baz',
```
Comment (jaymzh) [inline spec/svn_spec.rb]: probably want one for M and two spaces

File: lib/between_meals/repo/svn.rb (hunk 2 above)
Comment (odcinek) [inline lib/between_meals/repo/svn.rb]: That's what @elthariel pointed out above, and I agree with. We care about files changes (first column), and not attribute changes (second column). But space in the first column and whatever in second column is considered a file modification.

File: spec/svn_spec.rb (hunk 3 above)
Comment (odcinek) [inline spec/svn_spec.rb]: Added another case

**Pull request merged**